File: src/main/java/dev/slne/discord/config/ticket/TicketConfig.java
package dev.slne.discord.config.ticket;

import dev.slne.discord.config.BotConfig;
import lombok.Getter;
import lombok.ToString;
import org.spongepowered.configurate.objectmapping.ConfigSerializable;

import java.util.Map;

/**
 * The interface Ticket config.
 */
@ConfigSerializable
@Getter
@ToString
public class TicketConfig {

	private boolean enabled;
	private Map<String, TicketTypeConfig> ticketTypes;

	private TicketConfig() {
	}

	/**
	 * Gets config.
	 *
	 * @return the config
	 */
	public static TicketConfig getConfig() {
		return BotConfig.getConfig().getTicketConfig();
	}

}

File: src/main/kotlin/dev/slne/discord/config/ticket/TicketConfig.kt
package dev.slne.discord.config.ticket

import dev.slne.discord.config.botConfig
import org.spongepowered.configurate.objectmapping.ConfigSerializable

val ticketConfig: TicketConfig
    get() = botConfig.ticketConfig

@ConfigSerializable
class TicketConfig {

    var enabled = false
        private set

    lateinit var ticketTypes: Map<String, TicketTypeConfig>
}
